packages/clay-date-picker/src/TimePicker.tsx

```diff
@@ -35,27 +32,28 @@ const ClayDatePickerTimePicker: FunctionComponent<IProps> = ({
 	 * Handles the control time picker
 	 */
 	const handleOnChange = (values: Input) => {
-		const hours = values.hours === DEFAULT_VALUE ? 0 : Number(values.hours);
+		const hours =
+			values.hours === DEFAULT_VALUE
+				? DEFAULT_VALUE
+				: Number(values.hours);
 		const minutes =
-			values.minutes === DEFAULT_VALUE ? 0 : Number(values.minutes);
+			values.minutes === DEFAULT_VALUE
+				? DEFAULT_VALUE
+				: Number(values.minutes);
 
 		setValues(values);
 		onTimeChange(hours, minutes);
 	};
 
 	useEffect(() => {
-		const date = moment(currentTime, timeFormat);
-		const hours =
-			values.hours === DEFAULT_VALUE ? DEFAULT_VALUE : date.hour();
-		const minutes =
-			values.minutes === DEFAULT_VALUE ? DEFAULT_VALUE : date.minutes();
+		const time = currentTime.split(':');
 
 		setValues(prevValues => ({
 			...prevValues,
-			hours: String(hours),
-			minutes: String(minutes),
+			hours: String(time[0]),
+			minutes: String(time[1]),
```

Comment:
I'm a little hesitant at the back and forth use between `String()` and `Number()`. Any way we can keep just one type consistent?

Reply:
Yeah, I do not like it either, but it was necessary because in the end we're using the moment and the type is `number`, I could leave everything `string` to avoid it everywhere but I would have this problem there.

Reply:
ah gotcha, makes sense